sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/GroupByEndpointComponent.ts

```diff
@@ -37,55 +37,54 @@ export class GroupByEndpointComponent implements ExecutionContext {
       return { result: undefined, headers: getInitialHeader() };
     }
 
-    // Grab the next result
-    const { result, headers } = (await this.executionContext.nextItem()) as GroupByResponse;
+    const aggregateHeaders = getInitialHeader();
 
-    // If it exists, process it via aggreatators
-    if (result) {
-      const group = result.groupByItems ? await hashObject(result.groupByItems) : emptyGroup;
-      const aggergators = this.groupings.get(group);
-      const payload = result.payload;
-      if (aggergators) {
-        // Iterator over all results in the payload
-        Object.keys(payload).map((key) => {
-          const aggregateResult = extractAggergateResult(payload[key]);
-          aggergators.get(key).aggregate(aggregateResult);
-        });
-      } else {
-        // This is the first time we have seen a grouping. Setup the initial result without aggregate values
-        const grouping = new Map();
-        this.groupings.set(group, grouping);
-        // Iterator over all results in the payload
-        Object.keys(payload).map((key) => {
-          const aggregateType = this.queryInfo.groupByAliasToAggregateType[key];
-          // Create a new aggregator for this specific aggregate field
-          const aggreatator = createAggregator(aggregateType);
-          grouping.set(key, aggreatator);
-          if (aggregateType) {
+    while (this.executionContext.hasMoreResults()) {
+      // Grab the next result
+      const { result, headers } = (await this.executionContext.nextItem()) as GroupByResponse;
+      mergeHeaders(aggregateHeaders, headers);
+
+      // If it exists, process it via aggreatators
+      if (result) {
+        const group = result.groupByItems ? await hashObject(result.groupByItems) : emptyGroup;
+        const aggergators = this.groupings.get(group);
+        const payload = result.payload;
+        if (aggergators) {
+          // Iterator over all results in the payload
+          Object.keys(payload).map((key) => {
             const aggregateResult = extractAggergateResult(payload[key]);
-            aggreatator.aggregate(aggregateResult);
-          } else {
-            aggreatator.aggregate(payload[key]);
-          }
-        });
+            aggergators.get(key).aggregate(aggregateResult);
+          });
+        } else {
+          // This is the first time we have seen a grouping. Setup the initial result without aggregate values
+          const grouping = new Map();
+          this.groupings.set(group, grouping);
+          // Iterator over all results in the payload
+          Object.keys(payload).map((key) => {
```

Comment:
nit: `(key)` can be replaced by `key` or is there a style guildeline that requires the `()`?

Reply:
I think this is enforced by prettier so gonna leave it alone